File: tools/improve/tests/devstral-small-2/maudec/py/truth-table.py
#
#	<comment from="devstral-small-2">
#	The original code uses a match-case statement to implement logical AND and ANDC (AND with complement) operations. The logic can be simplified by directly using the built-in logical operators or by leveraging the fact that the cases are exhaustive and can be reduced to a single expression.
#	</comment>
#

def and_(a1: bool, a2: bool) -> bool:
    return a1 and a2

def andc(a1: bool, a2: bool) -> bool:
    return a1 and a2

File: tools/improve/tests/devstral-small-2/maudec/cpp/truth-table.cpp
//
//	<comment from="devstral-small-2">
//	The original code can be simplified by using the logical AND operator directly, as the conditions are equivalent to the standard boolean AND operation. The function `_and` and `andc` are essentially implementing the same logic as the `&&` operator in C++. The redundant checks can be removed, and the code can be simplified to a single return statement using the `&&` operator.
//	</comment>
//

bool _and(bool a1, bool a2) {
	return a1 && a2;
}

bool andc(bool a1, bool a2) {
	return a1 && a2;
}